index.bs

```diff
@@ -2214,24 +2214,41 @@ Note: Is is suggested that poses reported relative to a {{XRReferenceSpaceType/"
+A <dfn>Trusted UI</dfn> is an interface that the user can trust comes from the user agent, which the user may interact with without interference from the page. The user agent MUST support showing a [=trusted UI=]. Some form of [=trusted UI=] MUST be used to show permissions prompts.
+
+
+Broadly speaking, there are two options for user agents who wish to support [=trusted UI=]. One option is the <dfn>trusted immersive UI</dfn>, which is a [=trusted UI=] which does not exit immersive mode. It is tricky to design a good [=trusted immersive UI=] since the page can effectively draw any pixels it wishes to. User agents are not required to support [=trusted immersive UI=], they may instead temporarily pause/exit immersive mode and show non-immersive [=trusted UI=] to the user.
+
+A [=trusted UI=] MUST have the following properties:
+
+    - It must not be spoofable
+    - It indicates where the request/content displayed originates from
+    - If it relies on a shared secret with the user, this shared secret cannot be observed by a mixed reality capture (e.g. it may not be a gesture that can be seen by the camera)
+    - It is consistent between immersive experiences in the same UA
 
-Additionally, page content has the ability to make users uncomfortable in ways not related to performance. Badly applied tracking, strobing colors, and content intended to offend, frighten, or intimidate are examples of content which may cause the user to want to quickly exit the XR experience. Removing the XR device in these cases may not always be a fast or practical option. To accommodate this the user agent SHOULD provide users with an action, such as pressing a reserved hardware button or performing a gesture, that escapes out of WebXR content and displays the user agent's trusted UI.
+<div class="note">
+Note: Examples of [=trusted UI=] include:
+  - The default 2D mode browser shown when not in immersive mode
+  - A prompt shown within immersive mode which can only be interacted with via a reserved hardware button to prevent spoofing
+  - Pausing the immersive session and showing some form of native system environment in which a prompt can be shown
 
-When navigating between pages in XR the user agent should display trusted UI elements informing the user of the security information of the site they are navigating to which is normally presented by the 2D UI, such as the URL and encryption status.
+</div>
+
+In some cases it may be possible for a malicious page to infer what a user is typing on a virtual keyboard or how they are interacting with a virtual UI based solely on monitoring their head movements. This is especially true on devices that have limited input capabilities, such as Google Cardboard, which frequently require users to control a "gaze cursor" with their head orientation. For example: if not prevented from doing so a page could estimate what URL a user is entering into the user agent's URL bar by monitoring the user's interaction with the keyboard.
```

Comment:
I think this whole paragraph can probably be replaced with one sentence at the beginning of the next paragraph that says "Input snooping is bad, yo" because it isn't just gaze-based input that can be sniffed.  tracked pointers and screen are just as susceptible.